src/lib.rs

```diff
@@ -266,14 +283,182 @@ impl<R: Rng> Iterator for BlueNoise<R> {
+    fn distance(&self, point: Vec2, target: Vec2) -> f32 {
+        let diff = {
+            let tmp = (target - point).abs();
+            tmp.min(Vec2::new(self.0.width, self.0.height) - tmp)
+        };
+        diff.length_squared()
+    }
+
+    /// Check if a position is far enough away from
+    /// nearby previously created points.
+    fn is_valid(&self, point: Vec2) -> bool {
+        let x_range = {
+            let x = (point.x / self.0.cell_size) as isize;
+            ((x - 2)..(x + 3)).map(|x| x.rem_euclid(self.0.grid_width as isize) as usize)
+        };
+        let y_range = {
+            let y = (point.y / self.0.cell_size) as isize;
+            ((y - 2)..(y + 3)).map(|y| y.rem_euclid(self.0.grid_height as isize) as usize)
+        };
+
+        x_range.cartesian_product(y_range).all(|(x, y)| {
+            // if there is a point, check if it is further than our min radius
+            match self
+                .0
+                .grid
+                .get(y * self.0.grid_width + x)
+                .expect("Ended up out of bounds when fetching point.")
+            {
+                Some(target) => self.distance(point, *target) >= self.0.radius_squared,
+                None => true,
+            }
+        })
+    }
+
+    /// Get some nearby point
+    fn get_nearby(&mut self, position: Vec2, seed: f32, sample: u32) -> Vec2 {
+        let nearby = self.0.get_nearby(position, seed, sample);
+        Vec2::new(
+            nearby.x.rem_euclid(self.0.width),
+            nearby.y.rem_euclid(self.0.height),
+        )
+    }
+}
+
+impl<R: Rng> Iterator for WrappingBlueNoise<R> {
+    type Item = Vec2;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if !self.0.init {
+            self.0.init = true;
+            let x = self.0.rng.gen_range(0.0..self.0.width);
+            let y = self.0.rng.gen_range(0.0..self.0.height);
+            return Some(self.0.insert_point(Vec2::new(x, y)));
+        }
+
+        while !self.0.active_points.is_empty() {
+            let index = self.0.rng.gen::<f32>() * (self.0.active_points.len() - 1) as f32;
+            let parent = self.0.active_points[index as usize];
+
+            let seed = self.0.rng.gen::<f32>();
+            for sample in 0..self.0.max_samples {
+                let point = self.get_nearby(parent, seed, sample);
+                if self.is_valid(point) {
+                    return Some(self.0.insert_point(point));
+                }
+            }
+
+            self.0.active_points.remove(index as usize);
+        }
+
+        None
+    }
+}
+
 #[cfg(test)]
 mod test {
-    use crate::BlueNoise;
+    use crate::{BlueNoise, WrappingBlueNoise};
     use rand_pcg::Pcg64Mcg;
 
     #[test]
     fn get_points() {
         let noise = BlueNoise::<Pcg64Mcg>::new(100.0, 100.0, 1.0);
         assert!(noise.count() > 0);
+
+        let noise = WrappingBlueNoise::<Pcg64Mcg>::new(100.0, 100.0, 1.0);
+        assert!(noise.count() > 0);
```

Comment:
One miniscule thing, I'd prefer if we had two test functions. I know the tests are fairly pathetic atm anyways haha but at the least it would be nice to know which one fails.